### PR TITLE
Add Orchard 2^16 subtree roots to the coins db.

### PR DESCRIPTION
**Pull request by nuttycom**

### CODE REVIEW

File: src/rust/src/merkle_frontier.rs [hunk 1]
```diff
@@ -98,17 +99,45 @@ pub(crate) struct OrchardWallet;
 
 impl Orchard {
     /// Appends the note commitments in the given bundle to this frontier.
-    pub(crate) fn append_bundle(&mut self, bundle: &orchard_bundle::Bundle) -> bool {
+    pub(crate) fn append_bundle(
+        &mut self,
+        bundle: &orchard_bundle::Bundle,
+    ) -> Result<ffi::OrchardAppendResult, &'static str> {
```
Comment (daira) [inline src/rust/src/merkle_frontier.rs:105]: Is it possible to return a `Result<Option<[u8; 32]>, &'static str>` here?

Comment (nuttycom) [reply src/rust/src/merkle_frontier.rs:105]: I don't think so; there's no built-in binding for `Option`.

Comment (daira) [reply src/rust/src/merkle_frontier.rs:105]: Looks like cxx doesn't support it though. https://github.com/dtolnay/cxx/issues/87

**Review by daira — APPROVED** [review]
utACK 2ad7ab169126c28780c1f03db88eb9619f2053bf (only reviewed the last commit, because the rest is equivalent to #6691).

**Review by daira — DISMISSED** [review]
utACK b4ec4c5203b42b9a5f4dcfc16dfe2fdaeedd2039 (only reviewed the last commit, because the rest is equivalent to #6691).

**Review by ebfull — APPROVED** [review]

**Review by daira — APPROVED** [review]
utACK

**Pull request merged**